src/pocket/PocketApiGatewayLambdaIntegration.ts

```diff
@@ -0,0 +1,149 @@
+  private apiGatewayStage: APIGateway.ApiGatewayStage;
+
+  constructor(scope: Construct, name: string, config: PocketApiGatewayProps) {
+    super(scope, name);
+    this.apiGatewayRestApi = new APIGateway.ApiGatewayRestApi(
+      scope,
+      `api-gateway-rest`,
+      {
+        name: config.name,
+      }
+    );
+    // https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment
+    this.routes = this.createLambdaIntegrations(config);
+    const routeDependencies = this.routes.flatMap((route) => [
+      route.integration,
+      route.method,
+      route.resource,
+      route.lambda.lambda.versionedLambda,
+    ]);
+    // Deployment before adding permissions so we can restrict to the stage
+    this.apiGatewayDeployment = new APIGateway.ApiGatewayDeployment(
+      scope,
+      'api-gateway-deployment',
+      {
+        restApiId: this.apiGatewayRestApi.id,
+        // Removing the `id` from the resources in line 51 causes:
+        // `RangeError: Resolution error Maximum call stack size exceeded`
+        // However, this may not suffice for proper redeployment trigger...
+        // https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment#terraform-resources
+        triggers: {
+          redeployment: Fn.sha1(
+            Fn.jsonencode({
+              resources: routeDependencies.map((d) => d.id),
+              ...config,
+            })
+          ),
+        },
+        dependsOn: routeDependencies,
+      }
+    );
+    this.apiGatewayStage = new APIGateway.ApiGatewayStage(
+      scope,
+      'api-gateway-stage',
+      {
+        deploymentId: this.apiGatewayDeployment.id,
+        restApiId: this.apiGatewayRestApi.id,
+        stageName: config.stage,
+      }
+    );
+    this.addInvokePermissions();
+  }
+  private addInvokePermissions() {
+    this.routes.map(({ lambda, resource, method }) => {
+      const functionName = lambda.lambda.versionedLambda.functionName;
+      new LambdaFunction.LambdaPermission(
+        this,
+        `${functionName}-allow-gateway-lambda-invoke`,
+        {
+          functionName: lambda.lambda.versionedLambda.functionName,
+          action: 'lambda:InvokeFunction',
+          principal: 'apigateway.amazonaws.com',
+          // Grants access to invoke lambda on specified stage, method, resource path
+          // note the resource path has a leading `/`
+          sourceArn: `${this.apiGatewayRestApi.executionArn}/${this.apiGatewayStage.stageName}/${method.httpMethod}${resource.path}`,
+          qualifier: lambda.lambda.versionedLambda.name,
+        }
+      );
+    });
+  }
+  // loop over routes and generate methods
+  /**
+   * Generate aws proxy routes handled by aws lambda
+   * @param routes
+   */
+  private createLambdaIntegrations(config: PocketApiGatewayProps) {
+    return config.routes.map((route: ApiGatewayLambdaRoute) => {
+      const lambda = new PocketVersionedLambda(
```

Comment:
We should allow this lambda to accept the full config for the versioned lambda construct. We have a pattern around deploying lambdas with code deploy, we don't have to do that when we implement this, but we should have the option to set any allowed config.

With this implementation, we can't set alarms for example.

Reply:
Is that not `PocketVersionedLambdaProps`?